### PR TITLE
Add container status as property

### PR DESCRIPTION
**Pull request by asuresh4**

### CODE REVIEW

**Comment by asuresh4** [comment]
@keitwb - i've addressed the comment you made on https://github.com/signalfx/signalfx-agent/pull/982 about making these changes compatible with the service correlation feature you're working on

File: tests/monitors/kubernetes_cluster/kubernetes_cluster_test.py [hunk 1]
```diff
@@ -309,6 +309,15 @@ def test_containers(k8s_cluster):
                     container_id = container_status.container_id.replace("docker://", "").replace("cri-o://", "")
                     containers_cache[container_status.name] = container_id
                     assert wait_for(p(has_datapoint, agent.fake_services, dimensions={"container_id": container_id}))
+                    assert wait_for(
+                        p(
+                            has_dim_prop,
```
Comment (keitwb) [inline tests/monitors/kubernetes_cluster/kubernetes_cluster_test.py]: Be more specific here on the expected value of `container_status`.  Would be nice to test for the terminated/waiting states too but not sure how easy it is to force a container into that state when managed by Kubernetes.  If it isn't too convoluted to do, test for those other states as well.

Comment (asuresh4) [reply tests/monitors/kubernetes_cluster/kubernetes_cluster_test.py]: Updated the tests to pick up container status from the container objects retrieved. Let me know what you think.

File: internal/monitors/kubernetes/cluster/metrics/pods.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ import (
 	v1 "k8s.io/api/core/v1"
 )
 
-func datapointsForPod(pod *v1.Pod) []*datapoint.Datapoint {
+func datapointsForPod(pod *v1.Pod) ([]*datapoint.Datapoint, []*atypes.DimProperties) {
```
Comment (keitwb) [inline internal/monitors/kubernetes/cluster/metrics/pods.go]: This function should only return datapoints.  Why can't the dimProps stuff be done in the `dimPropsForPods` function below?

File: tests/monitors/kubernetes_cluster/kubernetes_cluster_test.py (hunk 1 above)
Comment (keitwb) [reply tests/monitors/kubernetes_cluster/kubernetes_cluster_test.py]: Seems good.

File: internal/monitors/kubernetes/cluster/metrics/pods.go (hunk 2 above)
Comment (asuresh4) [reply internal/monitors/kubernetes/cluster/metrics/pods.go]: `dimPropsForPods` does not have context about the pod object itself. It makes use of what we cache. I've moved this logic out to `handleAddPod` now.

**Comment by asuresh4** [comment]
Closing this as per discussion with @keitwb.

**Comment by asuresh4** [comment]
Changes LGTM but can't approve since I originally opened the PR.

**Pull request merged**